control-plane-operator/controllers/hostedcontrolplane/cloud/kubevirt/reconcile.go

```diff
@@ -0,0 +1,201 @@
+	ownerRef.ApplyTo(roleBinding)
+	roleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "Role",
+		Name:     role.Name,
+	}
+	roleBinding.Subjects = []rbacv1.Subject{
+		{
+			Namespace: sa.Namespace,
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      sa.Name,
+		},
+	}
+	return nil
+}
+
+func ReconcileDeployment(deployment *appsv1.Deployment, hcp *hyperv1.HostedControlPlane, serviceAccountName string, releaseImage *releaseinfo.ReleaseImage) error {
+	clusterName, ok := hcp.Labels["cluster.x-k8s.io/cluster-name"]
+	if !ok {
+		return fmt.Errorf("\"cluster.x-k8s.io/cluster-name\" label doesn't exist in HostedControlPlane")
+	}
+	deploymentConfig := newDeploymentConfig()
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: ccmLabels(),
+		},
+		Strategy: appsv1.DeploymentStrategy{
+			Type: appsv1.RecreateDeploymentStrategyType,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: ccmLabels(),
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					util.BuildContainer(CCMContainer(), buildCCMContainer(clusterName, releaseImage)),
+				},
+				Volumes:            []corev1.Volume{},
+				ServiceAccountName: serviceAccountName,
+			},
+		},
+	}
+
+	addVolumes(deployment)
+
+	config.OwnerRefFrom(hcp).ApplyTo(deployment)
+	deploymentConfig.ApplyTo(deployment)
+	return nil
+}
+
+func addVolumes(deployment *appsv1.Deployment) {
+
+	deployment.Spec.Template.Spec.Volumes = append(
+		deployment.Spec.Template.Spec.Volumes,
+		util.BuildVolume(ccmVolumeKubeconfig(), buildCCMVolumeKubeconfig),
+	)
+	deployment.Spec.Template.Spec.Volumes = append(
+		deployment.Spec.Template.Spec.Volumes,
+		util.BuildVolume(ccmCloudConfig(), buildCCMCloudConfig),
+	)
+}
+
+func getPodVolumeMounts() util.PodVolumeMounts {
+	return util.PodVolumeMounts{
+		CCMContainer().Name: util.ContainerVolumeMounts{
+			ccmVolumeKubeconfig().Name: "/etc/kubernetes/kubeconfig",
+			ccmCloudConfig().Name:      "/etc/cloud",
+		},
+	}
+}
+
+func buildCCMContainer(clusterName string, releaseImage *releaseinfo.ReleaseImage) func(c *corev1.Container) {
+	return func(c *corev1.Container) {
+		c.Image = releaseImage.ComponentImages()["kubevirt-cloud-controller-manager"]
+		c.ImagePullPolicy = corev1.PullIfNotPresent
+		c.Command = []string{"/bin/kubevirt-cloud-controller-manager"}
+		tenantClusterKubeconfig := "/etc/kubernetes/kubeconfig/kubeconfig"
+		c.Args = []string{
+			"--cloud-provider=kubevirt",
+			"--cloud-config=/etc/cloud/cloud-config",
+			"--kubeconfig", tenantClusterKubeconfig,
+			"--authentication-kubeconfig", tenantClusterKubeconfig,
+			"--authorization-kubeconfig", tenantClusterKubeconfig,
+			"--cluster-name", clusterName,
+		}
+		c.VolumeMounts = getPodVolumeMounts().ContainerMounts(c.Name)
+	}
+}
+
+func buildCCMVolumeKubeconfig(v *corev1.Volume) {
+	v.Secret = &corev1.SecretVolumeSource{
```

Comment:
You can use https://github.com/nirarg/hypershift/blob/3b74e616512d8ce3f5461af947f82e6ef6abaf0f/hack/tools/vendor/k8s.io/api/core/v1/types.go#L1184 so the mount point can be something like `/etc/kubernetes/kubeconfig/value` instead of redudant `/etc/kubernetes/kubeconfig/kubeconfig`
it would be something like 
```golang
items:
  kubeconfig: value
```

Reply:
I looked in other modules in hypershift, this is not used there
So lets keep it as it now